src/main/java/codeu/model/store/basic/IndexStore.java

```diff
@@ -66,13 +71,96 @@ public void addMapping (String word, Message message) {
   /**
     * Search function for a single word, returns a Set of messages from the Hashtable if the word is in it, otherwise return null
     */
-  public Set<Message> searchWord (String word) {
-    if (wordsHash.containsKey(word)) {
-      return wordsHash.get(word);
+  public List<Message> search (String searchTarget) {
+    if (searchTarget.contains("||")) {
+      return searchUnion(searchTarget);
+    }else if (searchTarget.contains("&&")) {
+      return searchIntersection(searchTarget);
+    }else if (wordsHash.containsKey(searchTarget)){
```

Comment:
Does it make sense to trim to first word here if the searchTarget has a whitespace?  The hash only contains single words right?

Reply:
Assuming if we search for "hello world" we'd rather at least return results for hello than nothing.
Alternatively, "hello world" could be parsed to automatically be "hello && world""